src/Server.kt

```diff
@@ -599,8 +625,11 @@ fun FskModel.run(simulationIndex: Int): String {
     }
 
     // 2. Simulation script
-    runSelectedSimulation(engine, simulationIndex)
-
+    if (userDefinedSim == null) {
```

Comment:
This null check can be avoided with the single simulation parameter. In that case only a call to *runSelectedSimulation(engine, simulation)* is needed.

Reply:
done